src/DynamicsMethods/SurfaceHoppingMethods/surface_hopping.jl

```diff
@@ -63,17 +63,29 @@ function rescale_velocity!(sim::AbstractSimulation{<:SurfaceHopping}, u)::Bool
     c = calculate_potential_energy_change(eigs, new_state, old_state)
 
     discriminant = b^2 - 4a * c
-    discriminant < 0 && return false # Frustrated hop with insufficient kinetic energy
-
-    root = sqrt(discriminant)
-    if b < 0
-        γ = (b + root) / 2a
-    else
-        γ = (b - root) / 2a
+    if discriminant < 0 # frustrated hop
+        if sim.method.rescaling == :standard
+            # Frustrated hop with insufficient kinetic energy, no velocity inversion
+            # Follow 1990 Tully recipe and do nothing
+            return false 
+        elseif sim.method.rescaling == :vinversion
+            # Frustrated hop with insufficient kinetic energy
+            # perform inversion of the velocity component along the nonadiabatic coupling vector
+            frustrated_hop_invert_velocity!(sim, DynamicsUtils.get_velocities(u), d)
+        else   
```

Comment:
In hindsight it would have been nice to implement this option using a struct instead of a symbol so we could use dispatch instead of an if statement but it's fine for now with only a few possible options.

Reply:
I did try to do that, but couldn't quite figure out how. If you have a moment, maybe you could post an Issue that describes the change and what benefits it might offer. Then it becomes something we could pick up at some point as we will surely run into limitations with the current approach as more methods get introduced.